query/physicalplan/aggregate.go

```diff
@@ -130,6 +135,7 @@ func NewHashAggregate(
 	aggregationFunction AggregationFunction,
 	columnToAggregate logicalplan.Expr,
 	groupByColumnMatchers []logicalplan.Expr,
+	finalStage bool,
```

Comment:
Instead of adding the final stage boolean to the physical plan execution, I think this logic should be part of the physical plan planning. i.e. when planning a multi-stage count, instead of planning a `count(final=true)` operator, you would just plan a `sum()` operator at the final stage.